README.md
# Cooming-Soon-Doodles

Download the project first. [Download](https://github.com/bleedingAyush/Cooming-Soon-Doodles/archive/refs/heads/main.zip) 


Then install yarn `npm install --global yarn`


Then go inside **bluetoof** folder and enter `yarn install`


After installation is complete, type `yarn start`, the server will start on **localhost:3000**


If you want to modify the Countdown Date, Go inside ComingSoon12.js and on line **124** modify the date.


If you want to modify the MailChimp url, Go inside ComingSoon12.js and on line **131** modify the url.

**Note**: The ComingSoon12.js file is inside src folder and inside pages folder.

For the production build type `yarn build`, a build folder will appear. You can publish the website, now with this folder.


changelog.txt
Date: 29 June, 2021
version: 2.0
===========================

1. React version updated to 17.0.2
2. Bootstrap version updated to 4.6.0



bluetoof/src/pages/NotFound.js
import React, { Fragment } from "react";
import { Link } from "react-router-dom";
import { Helmet } from "react-helmet";
import { ReactSVG } from "react-svg";

const NotFound = () => {
  return (
    <Fragment>
      <Helmet>
        <link
          href="https://fonts.googleapis.com/css2?family=Montserrat:wght@400;500;700&family=Roboto:wght@400;500&display=swap"
          rel="stylesheet"
        />
      </Helmet>
      <div
        className="cs-07-page-wrapper h-100 bg-img d-flex flex-column justify-content-between"
        style={{
          backgroundImage: `url(${process.env.PUBLIC_URL +
            "/assets/img/backgrounds/coming-soon-07.jpg"})`
        }}
      >
        {/*====================  header ====================*/}
        <header className="cs-07-header space-pt--30 space-pb--30">
          <div className="container">
            <div className="row">
              <div className="col-6">
                {/* logo */}
                <div className="cs-07-logo space-mb--35">
                  <Link to={process.env.PUBLIC_URL + "/"}>
                    <img
                      src={
                        process.env.PUBLIC_URL + "/assets/img/logo/logo-07.png"
                      }
                      className="img-fluid"
                      alt=""
                    />
                  </Link>
                </div>
              </div>
            </div>
          </div>
        </header>
        {/*====================  End of header  ====================*/}
        {/*====================  content ====================*/}
        <div className="cs-07-content">
          <div className="container">
            <div className="row">
              <div className="col-lg-10 ml-auto mr-auto">
                <h2 className="cs-07-content__title text-center">
                  Nothing Found Here
                </h2>

                <div className="cs-07-subscription-wrapper space-mt--50 text-center">
                  <div className="mc-newsletter-form">
                    <Link to={process.env.PUBLIC_URL + "/"}>
                      Go Back to Homepage
                    </Link>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        {/*====================  End of content  ====================*/}
        {/*====================  footer ====================*/}
        <footer className="cs-07-footer space-pt--25 space-pb--25">
          <div className="container">
            <div className="row align-items-center">
              <div className="col-12 col-md-6">
                {/* copyright */}
                <div className="cs-07-copyright">
                  &copy; {new Date().getFullYear() + " "}
                  <a
                    href="https://hasthemes.com/"
                    target="_blank"
                    rel="noopener noreferrer"
                  >
                    HasThemes
                  </a>
                  , all rights reserved
                </div>
              </div>
              <div className="col-12 col-md-6">
                {/* social icons */}
                <ul className="cs-07-social-icons d-flex align-items-center justify-content-center justify-content-md-end">
                  <li>
                    <a
                      href="http://www.facebook.com"
                      target="_blank"
                      rel="noopener noreferrer"
                    >
                      <ReactSVG
                        src={
                          process.env.PUBLIC_URL +
                          "/assets/img/icons/facebook-01.svg"
                        }
                      />
                    </a>
                  </li>
                  <li>
                    <a
                      href="http://www.twitter.com"
                      target="_blank"
                      rel="noopener noreferrer"
                    >
                      <ReactSVG
                        src={
                          process.env.PUBLIC_URL +
                          "/assets/img/icons/twitter-01.svg"
                        }
                      />
                    </a>
                  </li>
                  <li>
                    <a
                      href="http://www.instagram.com"
                      target="_blank"
                      rel="noopener noreferrer"
                    >
                      <ReactSVG
                        src={
                          process.env.PUBLIC_URL +
                          "/assets/img/icons/instagram-01.svg"
                        }
                      />
                    </a>
                  </li>
                  <li>
                    <a
                      href="http://www.linkedin.com"
                      target="_blank"
                      rel="noopener noreferrer"
                    >
                      <ReactSVG
                        src={
                          process.env.PUBLIC_URL +
                          "/assets/img/icons/linkedin-01.svg"
                        }
                      />
                    </a>
                  </li>
                  <li>
                    <a
                      href="http://www.youtube.com"
                      target="_blank"
                      rel="noopener noreferrer"
                    >
                      <ReactSVG
                        src={
                          process.env.PUBLIC_URL +
                          "/assets/img/icons/youtube-01.svg"
                        }
                      />
                    </a>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </footer>
        {/*====================  End of footer  ====================*/}
      </div>
    </Fragment>
  );
};

export default NotFound;


bluetoof/src/components/countdown/Renderer.js
import PropTypes from "prop-types";
import React, { Fragment } from "react";

const Renderer = ({ days, hours, minutes, seconds }) => {
  return (
    <Fragment>
      <div className="single-countdown">
        <span className="single-countdown__time">{days}</span>
        <span className="single-countdown__text">Days</span>
      </div>
      <div className="single-countdown">
        <span className="single-countdown__time">{hours}</span>
        <span className="single-countdown__text">Hours</span>
      </div>
      <div className="single-countdown">
        <span className="single-countdown__time">{minutes}</span>
        <span className="single-countdown__text">Minutes</span>
      </div>
      <div className="single-countdown">
        <span className="single-countdown__time">{seconds}</span>
        <span className="single-countdown__text">Seconds</span>
      </div>
    </Fragment>
  );
};

Renderer.propTypes = {
  days: PropTypes.number,
  hours: PropTypes.number,
  minutes: PropTypes.number,
  seconds: PropTypes.number
};

export default Renderer;


bluetoof/src/components/countdown/CircleCountdown.js
import React from "react";
import moment from "moment";

class CircleCountdown extends React.Component {
  state = {
    days: undefined,
    hours: undefined,
    minutes: undefined,
    seconds: undefined
  };

  componentDidMount() {
    this.interval = setInterval(() => {
      const { timeTillDate, timeFormat } = this.props;
      const then = moment(timeTillDate, timeFormat);
      const now = moment();
      const countdown = then.diff(now);

      let totalMilliseconds = moment.duration(countdown)._milliseconds;
      let seconds = Math.floor(totalMilliseconds / 1000);
      let minutes = Math.floor(seconds / 60);
      let hours = Math.floor(minutes / 60);
      let days = Math.floor(hours / 24);

      hours = hours - days * 24;
      minutes = minutes - days * 24 * 60 - hours * 60;
      seconds = seconds - days * 24 * 60 * 60 - hours * 60 * 60 - minutes * 60;

      days = days.toString();
      hours = hours.toString();
      minutes = minutes.toString();
      seconds = seconds.toString();

      this.setState({ days, hours, minutes, seconds });
    }, 1000);
  }

  componentWillUnmount() {
    if (this.interval) {
      clearInterval(this.interval);
    }
  }

  render() {
    const { days, hours, minutes, seconds } = this.state;

    // Mapping the date values to radius values
    const daysRadius = mapNumber(days, 99, 0, 0, 360);
    const hoursRadius = mapNumber(hours, 24, 0, 0, 360);
    const minutesRadius = mapNumber(minutes, 60, 0, 0, 360);
    const secondsRadius = mapNumber(seconds, 60, 0, 0, 360);

    if (!seconds) {
      return null;
    }

    return (
      <div className="countdown-wrapper">
        {days && (
          <div className="countdown-item">
            <SVGCircle radius={daysRadius} />
            <div className="countdown-item__text">
              {days}
              <span>days</span>
            </div>
          </div>
        )}
        {hours && (
          <div className="countdown-item">
            <SVGCircle radius={hoursRadius} />
            <div className="countdown-item__text">
              {hours}
              <span>hours</span>
            </div>
          </div>
        )}
        {minutes && (
          <div className="countdown-item">
            <SVGCircle radius={minutesRadius} />
            <div className="countdown-item__text">
              {minutes}
              <span>minutes</span>
            </div>
          </div>
        )}
        {seconds && (
          <div className="countdown-item">
            <SVGCircle radius={secondsRadius} />
            <div className="countdown-item__text">
              {seconds}
              <span>seconds</span>
            </div>
          </div>
        )}
      </div>
    );
  }
}

const SVGCircle = ({ radius }) => (
  <svg className="countdown-svg">
    <path
      fill="none"
      stroke="#FF6B6B"
      strokeWidth="25"
      d={describeArc(70, 70, 68, 0, radius)}
    />
  </svg>
);

function polarToCartesian(centerX, centerY, radius, angleInDegrees) {
  var angleInRadians = ((angleInDegrees - 90) * Math.PI) / 180.0;

  return {
    x: centerX + radius * Math.cos(angleInRadians),
    y: centerY + radius * Math.sin(angleInRadians)
  };
}

function describeArc(x, y, radius, startAngle, endAngle) {
  var start = polarToCartesian(x, y, radius, endAngle);
  var end = polarToCartesian(x, y, radius, startAngle);

  var largeArcFlag = endAngle - startAngle <= 180 ? "0" : "1";

  var d = [
    "M",
    start.x,
    start.y,
    "A",
    radius,
    radius,
    0,
    largeArcFlag,
    0,
    end.x,
    end.y
  ].join(" ");

  return d;
}

function mapNumber(number, in_min, in_max, out_min, out_max) {
  return (
    ((number - in_min) * (out_max - out_min)) / (in_max - in_min) + out_min
  );
}

export default CircleCountdown;


bluetoof/src/components/newsletter/SubscribeEmail.js
import PropTypes from "prop-types";
import React, { Fragment } from "react";
import MailchimpSubscribe from "react-mailchimp-subscribe";

const CustomForm = ({ status, message, onValidated }) => {
  let email;
  const submit = () => {
    email &&
      email.value.indexOf("@") > -1 &&
      onValidated({
        EMAIL: email.value
      });
    email.value = "";
  };

  return (
    <Fragment>
      <div className="mc-newsletter-form space-mb--10">
        <input
          ref={node => (email = node)}
          type="email"
          placeholder="Enter email here"
          required
        />
        <button onClick={submit}>NOTIFY ME</button>
      </div>

      <div className="mailchimp-alerts">
        {status === "sending" && (
          <div className="mailchimp-submitting">sending...</div>
        )}
        {status === "error" && (
          <div
            className="mailchimp-error"
            dangerouslySetInnerHTML={{ __html: message }}
          />
        )}
        {status === "success" && (
          <div
            className="mailchimp-success"
            dangerouslySetInnerHTML={{ __html: message }}
          />
        )}
      </div>
    </Fragment>
  );
};

const SubscribeEmail = ({ mailchimpUrl }) => {
  return (
    <div>
      <MailchimpSubscribe
        url={mailchimpUrl}
        render={({ subscribe, status, message }) => (
          <CustomForm
            status={status}
            message={message}
            onValidated={formData => subscribe(formData)}
          />
        )}
      />
    </div>
  );
};

SubscribeEmail.propTypes = {
  mailchimpUrl: PropTypes.string
};

export default SubscribeEmail;


bluetoof/src/pages/comingSoon12.js
import React, { Fragment } from "react";
import { Link } from "react-router-dom";
import { Helmet } from "react-helmet";
import { ReactSVG } from "react-svg";
import Countdown from "react-countdown";
import Renderer from "../components/countdown/Renderer";
import SubscribeEmail from "../components/newsletter/SubscribeEmail";

const ComingSoon12 = () => {
  return (
    <Fragment>
      <Helmet>
        <style>
          {`
          
          @font-face {
              font-family: "Steelfish Rg";
              font-weight: bold;
              font-style: normal;
          
              src: url(${
                process.env.PUBLIC_URL +
                "/assets/fonts/SteelfishRg-Bold/SteelfishRg-Bold.eot"
              });
              src: url(${
                process.env.PUBLIC_URL +
                "/assets/fonts/SteelfishRg-Bold/SteelfishRg-Bold.eot?#iefix"
              }) format("embedded-opentype"),
              url(${
                process.env.PUBLIC_URL +
                "/assets/fonts/SteelfishRg-Bold/SteelfishRg-Bold.woff2"
              }) format("woff2"),
              url(${
                process.env.PUBLIC_URL +
                "/assets/fonts/SteelfishRg-Bold/SteelfishRg-Bold.woff"
              }) format("woff"),
              url(${
                process.env.PUBLIC_URL +
                "/assets/fonts/SteelfishRg-Bold/SteelfishRg-Bold.ttf"
              }) format("truetype"),
              url(${
                process.env.PUBLIC_URL +
                "/assets/fonts/SteelfishRg-Bold/SteelfishRg-Bold.svg#SteelfishRg-Bold"
              }) format("svg");
          }

          @font-face {
            font-family: "Steelfish Eb";
            font-weight: 900;
            font-style: normal;
            src: url(${
              process.env.PUBLIC_URL +
              "/assets/fonts/SteelfishEb-Regular/SteelfishEb-Regular.eot"
            });
            src: url(${
              process.env.PUBLIC_URL +
              "/assets/fonts/SteelfishEb-Regular/SteelfishEb-Regular.eot?#iefix"
            }) format('embedded-opentype'),
                url(${
                  process.env.PUBLIC_URL +
                  "/assets/fonts/SteelfishEb-Regular/SteelfishEb-Regular.woff2"
                }) format('woff2'),
                url(${
                  process.env.PUBLIC_URL +
                  "/assets/fonts/SteelfishEb-Regular/SteelfishEb-Regular.woff"
                }) format('woff'),
                url(${
                  process.env.PUBLIC_URL +
                  "/assets/fonts/SteelfishEb-Regular/SteelfishEb-Regular.ttf"
                }) format('truetype'),
                url(${
                  process.env.PUBLIC_URL +
                  "/assets/fonts/SteelfishEb-Regular/SteelfishEb-Regular.svg#SteelfishEb-Regular"
                }) format('svg');
          }
        
          `}
        </style>
      </Helmet>
      <div
        className="cs-12-page-wrapper h-100 bg-img d-flex flex-column justify-content-between"
        style={{
          backgroundImage: `url(${
            process.env.PUBLIC_URL +
            "/assets/img/backgrounds/coming-soon-12.jpg"
          })`,
        }}
      >
        {/*====================  header ====================*/}
        <header className="cs-12-header space-pt--30 space-pb--30" />
        {/*====================  End of header  ====================*/}
        {/*====================  content ====================*/}
        <div className="cs-12-content">
          <div className="container">
            <div className="row">
              <div className="col-6 ml-auto mr-auto">
                {/* logo */}
                <div className="cs-12-logo text-center space-mb--35">
                  <Link to={process.env.PUBLIC_URL + "/"}>
                    <h2
                      className="cs-12-content__ text-center"
                      style={{ color: "white" }}
                    >
                      America Doodles
                    </h2>
                    {/* <img
                      src={
                        process.env.PUBLIC_URL + "/assets/img/logo/logo-12.png"
                      }
                      className="img-fluid"
                      alt=""
                    /> */}
                  </Link>
                </div>
              </div>
            </div>
            <div className="row">
              <div className="col-lg-10 ml-auto mr-auto">
                <h2 className="cs-12-content__title text-center">
                  WE ARE COMING SOON
                </h2>
                {/* countdown */}
                <div className="cs-12-countdown space-mt--50">
                  <Countdown
                    date={new Date("December 20, 2022 12:12:00")}
                    renderer={Renderer}
                  />
                </div>
                <div className="cs-12-subscription-wrapper space-mt--50 text-center">
                  {/* subscribe email */}
                  <SubscribeEmail mailchimpUrl="https://devitems.us11.list-manage.com/subscribe/post?u=6bbb9b6f5827bd842d9640c82&amp;id=05d85f18ef" />
                </div>
              </div>
            </div>
          </div>
        </div>
        {/*====================  End of content  ====================*/}
        {/*====================  footer ====================*/}
        <footer className="cs-12-footer space-pt--25 space-pb--25">
          <div className="container">
            <div className="row align-items-center">
              <div className="col-12 col-md-6">
                {/* copyright */}
                <div className="cs-12-copyright">
                  &copy; {new Date().getFullYear() + " "}
                  <a
                    href="https://hasthemes.com/"
                    target="_blank"
                    rel="noopener noreferrer"
                  >
                    AmericaDoodles
                  </a>
                  , all rights reserved
                </div>
              </div>
              <div className="col-12 col-md-6">
                {/* social icons */}
                <ul className="cs-12-social-icons d-flex align-items-center justify-content-center justify-content-md-end">
                  <li>
                    <a href="" target="_blank" rel="noopener noreferrer">
                      <ReactSVG
                        src={
                          process.env.PUBLIC_URL +
                          "/assets/img/icons/facebook-04.svg"
                        }
                      />
                    </a>
                  </li>
                  <li>
                    <a href="" target="_blank" rel="noopener noreferrer">
                      <ReactSVG
                        src={
                          process.env.PUBLIC_URL +
                          "/assets/img/icons/twitter-04.svg"
                        }
                      />
                    </a>
                  </li>
                  <li>
                    <a href="" target="_blank" rel="noopener noreferrer">
                      <ReactSVG
                        src={
                          process.env.PUBLIC_URL +
                          "/assets/img/icons/linkedin-04.svg"
                        }
                      />
                    </a>
                  </li>
                  <li>
                    <a href="" target="_blank" rel="noopener noreferrer">
                      <ReactSVG
                        src={
                          process.env.PUBLIC_URL +
                          "/assets/img/icons/youtube-04.svg"
                        }
                      />
                    </a>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </footer>
        {/*====================  End of footer  ====================*/}
      </div>
    </Fragment>
  );
};

export default ComingSoon12;
